Axel

hauduc opened the pull request:
Added a few clarifying points on the Windows page.

Sorry about removing pull request #12, didn't know I could delete your comments...

hauduc commented:
Also, just thought of this but is it still possible to add the slides (/slides/tutorials/2021/bash.pdf)) within the worksheet webpage, or somewhere else on the website as well?

stephan-koenig commented:
Hi Axel,

I just pushed updates to the `main` branch. On your local machine, please pull the main:

```
git switch main
git pull
```

Then merge the main into `axel` and push:

```
git switch axel
git merge main
git push
```

I renamed your slides and such, and we need to merge the changes from `main` back into `axel` so everything syncs up. (In general, it is good practice to merge `main` back into your branch before making a new pull request. In that case, I hadn't had time to push the latest commits.)

Thanks,

Stephan

stephan-koenig commented:
> Also, just thought of this but is it still possible to add the slides (/slides/tutorials/2021/bash.pdf)) within the worksheet webpage, or somewhere else on the website as well?

Yes, I will implement this today.

stephan-koenig commented:
From #13:
> Hi Stephan, 
> 
> PowerShell will behave identically whether accessed through the standalone Windows app or through Windows Terminal though no? I don't want to make things too complicated because I had students with issues with Windows Terminal last year. If they can open Command Prompt or PowerShell on the standalone apps, they can use ssh and I just wanted to emphasize to then that that is sufficient. 
> 
> Will be careful with which files to commit in the future. 
> 
> What was the issue with the rendering of _main? I just saw that the bookdown is deployed though, looks great! 
> 
> Best, 
> Axel

Ah, interesting. What were the issues with Windows Terminal?

In the end, you will have to deal with most of the technical issues at the front line, but PowerShell was causing all kinds of problems (although I can't remember them right now). Do you think the command prompt will work reliably? If so, then let's only mention that.

stephan-koenig commented:
I should also explain that I removed the `2020` and `2021` directories because you will replace the file in place (i.e. you will replace the 2020 version of `scripts.pdf` with the 2021 version. One purpose of version tracking is to avoid any manual tracking of versions in file/directory names.

hauduc commented:
Hi Stephan,

I'm having trouble with the commands you gave me:

axel@Axel-Surface MINGW64 ~/Documents/github/MICB405 (axel)
$ git switch main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.

axel@Axel-Surface MINGW64 ~/Documents/github/MICB405 (main)
$ git pull
Already up to date.

axel@Axel-Surface MINGW64 ~/Documents/github/MICB405 (main)
$ git switch axel
Switched to branch 'axel'
Your branch is up to date with 'origin/axel'.

axel@Axel-Surface MINGW64 ~/Documents/github/MICB405 (axel)
$ git merge main
Auto-merging Rmds/windows.Rmd
CONFLICT (content): Merge conflict in Rmds/windows.Rmd
Automatic merge failed; fix conflicts and then commit the result.

stephan-koenig commented on Rmds/bash.Rmd:
```diff
@@ -18,26 +14,27 @@ Use the following command, replacing `<username>` with your actual username:
+Enter your password when prompted. Don't worry when characters don't appear - this is an intentional security feature.
 
-Let’s break this command down: `ssh` is the command and stands for "secure-shell". It allows users to log on to remote (opposite of "local", which is your computer in this instance) servers. All following text is the argument. There may be many arguments and each of these would be separated by spaces. `<username>` is mostly obvious, but crucially this positions your shell in the `home` directory  ...
+You may have to respond to additonal prompts by typing in an answer (such as `yes`) and pressing <kbd>Enter</kbd>
```
```suggestion
You may have to respond to additonal prompts by typing in an answer (such as `yes`) and pressing <kbd>Enter</kbd>.
```

hauduc commented:
It says that there's a conflict and it doesn't want to merge axel

stephan-koenig commented:
That will happen a lot. Here is a link on how to fix merge conflicts: https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts

hauduc commented:
Thank you. I committed the resolved merge.

Best,
Axel
________________________________
From: Stephan Koenig ***@***.***>
Sent: Wednesday, September 8, 2021 13:10
To: EDUCE-UBC/MICB405 ***@***.***>
Cc: Axel Hauduc ***@***.***>; Author ***@***.***>
Subject: Re: [EDUCE-UBC/MICB405] Axel (#14)


That will happen a lot. Here is a link on how to fix merge conflicts: https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.atlassian.com%2Fgit%2Ftutorials%2Fusing-branches%2Fmerge-conflicts&data=04%7C01%7C%7Cd1e5188d02c14b37d8b508d97304a955%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667286124347379%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cM7LH181kpl0qel%2Bp9tESMrONxlArjoFq9cPffKoC0k%3D&reserved=0>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FEDUCE-UBC%2FMICB405%2Fpull%2F14%23issuecomment-915535584&data=04%7C01%7C%7Cd1e5188d02c14b37d8b508d97304a955%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667286124357381%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=57geId33Wjb%2FhXIfh6%2F1VrNS6uoRoQvF7OktkcqPNoo%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FANMIURWRJT5WRU32U3PEFTTUA67KDANCNFSM5DVRAG4A&data=04%7C01%7C%7Cd1e5188d02c14b37d8b508d97304a955%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667286124357381%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=B4dp3TxFvpEgixNh5Tn%2FE3K%2Fy5mEvsxkirRyAY%2FHLB0%3D&reserved=0>.

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,19 @@ These instructions will ensure that your local computer can connect to a Linux s
 -   OpenSSH comes pre-installed if you have Windows 10 version 1803 or newer. If you updated to the latest Windows 10 version, you should already have OpenSSH available to you. If your Windows 10 version is older because you cannot update, follow the [official Microsoft documentation on installing OpenSSH Client](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_insta ...
 
--   The simplest way to access OpenSSH on Windows is to open the "Command Prompt" application, discoverable through the search bar. This application will provide sufficient functionality for all server-based work you will do in MICB405.
+-   The simplest way to access OpenSSH on Windows is to open either the "Command Prompt" or "PowerShell" applications, discoverable through the search bar. These applications will provide sufficient functionality for all server-based work you will do in MICB405.
```
Let's decide on only one app—no need to give options. If students know about options themselves, they can make these decisions on their own.

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,19 @@ These instructions will ensure that your local computer can connect to a Linux s
 ## Additional considerations for Windows {-}
 
-Be aware that when you open a new Windows Terminal window that you will initially use an entirely different shell/language (called [PowerShell](https://docs.microsoft.com/en-us/powershell/)) on your computer. Only after you are connected to the server (using OpenSSH, see above) will you be using Bash. Tread with caution if executing commands other than `ssh` or `scp` from the tutorials while stil ...
+You can make your command line experience more modern and customizeable on Windows with the use of [Windows Terminal](https://docs.microsoft.com/en-us/windows/terminal/get-started). 
+
```
I would remove this if you do not want students to use Windows Terminal in general. Let's stick with one option. Don't know if you saw my other [comment](https://github.com/EDUCE-UBC/MICB405/pull/14#issuecomment-915524078). Whatever you decide, then all references should talk about that app, e.g. "command prompt".

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,19 @@ These instructions will ensure that your local computer can connect to a Linux s
 ## Optional: Windows Subsystem for Linux
 
-You can create a true Linux environment at the command line by setting up [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10). You can then access Bash of this virtualized Linux to practice and run Linux commands on your machine without connecting to the server.
+You can create a true Linux environment at the command line by setting up [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10). You can then access Bash on a virtualized Linux distribution, either directly through that distribution's Windows app or on Windows Terminal, to practice and run Linux commands on your machine without connecting to the server.
```
Again, I would not mention Windows terminal if that's not our default.

hauduc replied:
Sure, sounds good. The good thing about the in-person tutorials is that at least if someone is super confused I can help them

stephan-koenig commented:
> Thank you. I committed the resolved merge. Best, Axel
 Awesome! Merge conflicts still freak me out :grin:

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,22 @@ These instructions will ensure that your local computer can connect to a Linux s
 
--   The simplest way to access OpenSSH on Windows is to open the "Command Prompt" application, discoverable through the search bar. This application will provide sufficient functionality for all server-based work you will do in MICB405.
+
+-   The simplest way to access OpenSSH on Windows is to open either the "Command Prompt" or "PowerShell" applications, discoverable through the search bar. These applications will provide sufficient functionality for all server-based work you will do in MICB405.
```
```suggestion
-   The simplest way to access OpenSSH on Windows is to open the "Command Prompt" application, discoverable through the search bar. This application will provide sufficient functionality for all server-based work you will do in MICB405.
```

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,22 @@ These instructions will ensure that your local computer can connect to a Linux s
 ## Additional considerations for Windows {-}
 
-Be aware that when you open a new Windows Terminal window that you will initially use an entirely different shell/language (called [PowerShell](https://docs.microsoft.com/en-us/powershell/)) on your computer. Only after you are connected to the server (using OpenSSH, see above) will you be using Bash. Tread with caution if executing commands other than `ssh` or `scp` from the tutorials while stil ...
+You can make your command line experience more modern and customizeable on Windows with the use of [Windows Terminal](https://docs.microsoft.com/en-us/windows/terminal/get-started). 
+
```
```suggestion
```

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,22 @@ These instructions will ensure that your local computer can connect to a Linux s
-Be aware that when you open a new Windows Terminal window that you will initially use an entirely different shell/language (called [PowerShell](https://docs.microsoft.com/en-us/powershell/)) on your computer. Only after you are connected to the server (using OpenSSH, see above) will you be using Bash. Tread with caution if executing commands other than `ssh` or `scp` from the tutorials while stil ...
+You can make your command line experience more modern and customizeable on Windows with the use of [Windows Terminal](https://docs.microsoft.com/en-us/windows/terminal/get-started). 
+
+Be aware that when you open Command Prompt or Powershell, either directly through the associated app or in a new Windows Terminal tab, that you are using an entirely different local shell [with its own set of commands](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/windows-commands) unique from Linux. Only after you are connected to the server (using OpenSSH, see  ...
```
```suggestion
Be aware that when you open Command Prompt, you are using an entirely different local shell [with its own set of commands](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/windows-commands) unique from Linux. Only after you are connected to the server (using OpenSSH, see above) will you be using Bash. Tread with caution if executing commands other than `ssh` or `scp` from the tutorials while still in Command Prompt (i.e. while not connected to the server).
```

stephan-koenig commented on Rmds/windows.Rmd:
```diff
@@ -24,17 +24,22 @@ These instructions will ensure that your local computer can connect to a Linux s
 ## Optional: Windows Subsystem for Linux
 
-You can create a true Linux environment at the command line by setting up [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10). You can then access Bash of this virtualized Linux to practice and run Linux commands on your machine without connecting to the server.
+You can create a true Linux environment at the command line by setting up [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10). You can then access Bash on a virtualized Linux distribution, either directly through that distribution's Windows app or on Windows Terminal, to practice and run Linux commands on your machine without connecting to the server.
```
```suggestion
You can create a true Linux environment at the command line by setting up [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10). You can then access Bash on a virtualized Linux distribution, either directly through that distribution's Windows app or on Command Prompt to practice and run Linux commands on your machine without connecting to the server.
```
